jni_bindings/src/lib.rs

```diff
@@ -278,176 +341,192 @@ fn from_attr(attr: &dom::Attribute) -> Attribute {
-            0
-        }
     }
 }
 
 #[no_mangle]
-pub extern "system" fn Java_org_phoenixframework_liveview_lib_Document_get_1node_1element(
-    env: JNIEnv,
+pub unsafe extern "system" fn Java_org_phoenixframework_liveview_lib_Document_get_1node_1element(
+    mut env: JNIEnv,
     _: JClass,
+    // non-null raw pointer to node
     node: jlong,
+    // Raw pointer to Element
 ) -> jlong {
-    let mut clone = unsafe { env.unsafe_clone() };
-    let result = catch_unwind(|| unsafe {
-        let node: *mut Node = jlong_to_pointer::<Node>(node).as_mut().unwrap();
-        let ret = Box::new((*node).data.element);
-        Box::into_raw(ret) as jlong as jlong
-    });
-
-    match result {
-        Ok(jlong) => jlong,
-        Err(err) => {
-            let exception_class = clone.find_class("java/lang/RuntimeException").unwrap();
-            let message = format!("Rust panic occurred in get_node_element: {:?}", err);
-            clone.throw_new(exception_class, message).unwrap();
-            0
-        }
+    let node = node as *mut Node;
+
+    if node.is_null() {
+        env.throw_new(
+            "java/lang/NullPointerException",
+            "Document::get_node_element called with `node` as null pointer",
+        )
+        .unwrap();
+
+        0
+    } else {
+        let node = &*node;
+        let element = Box::new(node.data.element.clone());
+
+        Box::into_raw(element) as jlong
     }
 }
 
 #[no_mangle]
-pub extern "system" fn Java_org_phoenixframework_liveview_lib_Node_00024Element_drop(
-    env: JNIEnv,
+pub unsafe extern "system" fn Java_org_phoenixframework_liveview_lib_Node_00024Element_drop(
+    mut env: JNIEnv,
     _: JClass,
+    // non-null raw pointer to Element
     this: jlong,
 ) {
-    let mut clone = unsafe { env.unsafe_clone() };
-    let result = catch_unwind(|| unsafe {
-        let dom: *mut Element = jlong_to_pointer::<Element>(this).as_mut().unwrap();
-        let dom = Box::from_raw(dom);
-        drop(dom);
-    });
-
-    if let Err(err) = result {
-        let exception_class = clone.find_class("java/lang/RuntimeException").unwrap();
-        let message = format!("Rust panic occurred in Element drop: {:?}", err);
-        clone.throw_new(exception_class, message).unwrap();
+    let this = this as *mut Element;
+
+    if this.is_null() {
+        env.throw_new(
+            "java/lang/NullPointerException",
+            "Node.Element::drop called with null pointer",
+        )
+        .unwrap();
+    } else {
+        let _element = Box::from_raw(this);
     }
 }
 
 #[no_mangle]
-pub extern "system" fn Java_org_phoenixframework_liveview_lib_Document_get_1node_1element_1namespace<
+pub unsafe extern "system" fn Java_org_phoenixframework_liveview_lib_Node_00024Element_get_1namespace<
     'local,
 >(
-    env: JNIEnv<'local>,
+    mut env: JNIEnv<'local>,
     _: JClass<'local>,
-    element: jlong,
+    // non-null raw pointer to Element
```

Comment:
The `non-null` can't be true or there would be no need for the `this.is_null()` check below.
```suggestion
    // raw pointer to Element
```